File: backend/node/src/aws/s3.mjs
import config from "../config.mjs";
import { S3Client, PutObjectCommand, DeleteObjectCommand } from "@aws-sdk/client-s3";

//recibe un buffer y lo sube a S3
const uploadImageS3 = async (buff, path) => {
    const client = new S3Client({
      region: config.region,
      credentials: {
        accessKeyId: config.accessKeyId,
        secretAccessKey: config.secretAccessKey,
      },
    });
  
    const command = new PutObjectCommand({
      Bucket: config.bucket,
      Key: path,
      Body: buff,
      ContentType: "image/jpeg",
    });
  
    try {
      const response = await client.send(command);
      return response;
    } catch (error) {
      console.error(error);
      return null;
    }finally{
      client.destroy();
    }
  };

  const deleteObjectS3 = async (path) => {
    let parts = path.split('/');
    parts = parts[parts.length - 2] + "/" + parts[parts.length - 1];
  
    const client = new S3Client({
      region: config.region,
      credentials: {
        accessKeyId: config.accessKeyId,
        secretAccessKey: config.secretAccessKey,
      },
    });
    const command = new DeleteObjectCommand({
      Bucket: config.bucket,
      Key: parts,
    });
  
    try {
      const response = await client.send(command);
      return response;
    } catch (error) {
      console.error(error);
      return null;
    }
  };
  
  export { uploadImageS3, deleteObjectS3};
  

File: backend/python/src/aws/s3.py
from config import config
import boto3
from botocore.exceptions import NoCredentialsError, PartialCredentialsError

def uploadImageS3(buff, path):
    #Configuracion del cliente S3
    s3_client = boto3.client(
        's3',
        region_name=config['region'],
        aws_access_key_id=config['accessKeyId'],
        aws_secret_access_key=config['secretAccessKey']
    )

    try:
        response = s3_client.put_object(
            Body=buff,
            Bucket=config['bucket'],
            Key=path,
            ContentType='image/jpeg'
        )

        return response
    except (NoCredentialsError, PartialCredentialsError) as e:
        print("Error al subir la imagen a S3" + str(e))
        return None
    except Exception as e:
        print("Error al subir la imagen a S3" + str(e))
        return None

def deleteObjectS3(path):
    
    try:
        parts = path.split("/")
        parts = parts[-2] + "/" + parts[-1]

        #Configuracion del cliente S3
        s3_client = boto3.client(
            's3',
            region_name=config['region'],
            aws_access_key_id=config['accessKeyId'],
            aws_secret_access_key=config['secretAccessKey']
        )

        response = s3_client.delete_object(
            Bucket=config['bucket'],
            Key=parts
        )

        return response
    except (NoCredentialsError, PartialCredentialsError) as e:
        print("Error al eliminar la imagen de S3" + str(e))
        return None
    except Exception as e:
        print("Error al eliminar la imagen de S3" + str(e))
        return None

 